draft-ietf-ppm-dap.md

```diff
@@ -2653,58 +2638,60 @@ running aggregation jobs ({{aggregate-flow}}) until the Collector initiates a
 collection job. This is because, in general, the aggregation parameter is not
 known until this point. In certain situations it is possible to predict the
 aggregation parameter in advance. For example, for Prio3 the only valid
-aggregation parameter is the empty string. For these reasons, the collection job
-is handled asynchronously. If aggregation is performed eagerly, the Leader MUST
-validate that the aggregation parameter received in the `CollectionJobReq`
-matches the aggregation parameter used in aggregations.
+aggregation parameter is the empty string.
 
 Upon receipt of a `CollectionJobReq`, the Leader begins by checking that it
-recognizes the task ID in the request path. If not, it MUST abort with error
-`unrecognizedTask`.
+recognizes the task ID. If not, it MUST abort with error `unrecognizedTask`.
+
+Next, the Leader checks that the indicated batch mode matches the task's batch
+mode. If not, then the Leader MUST abort with error `invalidMessage`.
 
 Next, the Leader checks that the aggregation parameter is valid as described in
-{{agg-param-validation}}. If the aggregation parameter is invalid, then the
-Leader MUST abort with error `invalidAggregationParameter`.
+{{agg-param-validation}}. If not, then the Leader MUST abort with error
+`invalidAggregationParameter`.
+
+If aggregation is performed eagerly, the Leader MUST validate that the
+aggregation parameter received in the `CollectionJobReq` matches the aggregation
+parameter used in aggregations.
+
+Changing a collection job's parameters is illegal, so if there are further PUT
+requests to the collection job with a different `CollectionJobReq`, the Leader
+MUST abort with a client error.
```

Comment:
What is a "client error"?

Reply:
HTTP status in the 400 range. I think it's clear enough from the context that we're discussing HTTP requests, and we lay out what it means to "abort with..." in [the section on errors](https://ietf-wg-ppm.github.io/draft-ietf-ppm-dap/draft-ietf-ppm-dap.html#section-3.4).

Reply:
I guess what I'm asking for is to be prescriptive about what error to use from that table.

Reply:
OK, I think that's fair. I also need to do a pass here to make sure we distinguish clearly between those conditions that get checked during the handling of the PUT request regardless of whether the job will be run asynchronously (so stuff like "is the message well-formed" that's cheap) vs. checks that you'd want to defer (so stuff like "is the right agg param used" that requires fetching a bunch of data from storage).